### PR TITLE
Sets the MTU when a non-default value is specified

### PR DESCRIPTION
**Pull request by jterry75**
Resolves: #115

### CODE REVIEW

**Comment by jterry75** [comment]
Don't merge yet. Need to get a build from jenkins and test

File: service/gcsutils/gcstools/netnsConfig.go [hunk 1]
```diff
@@ -98,6 +98,13 @@ func netnsConfig() error {
 		return fmt.Errorf("netlink.LinkByName(%s) failed: %v", *ifStr, err)
 	}
 
+	// User requested non-default MTU size
+	if a.EncapOverhead != 0 {
+		if err = netlink.LinkSetMTU(link, int(a.EncapOverhead)); err != nil {
```
Comment (rn) [inline service/gcsutils/gcstools/netnsConfig.go]: I would have not expected from the name `EncapOverhead` that it actually contains the MTU. If `EncapOverhead` encodes the overhead required for encapsulation, I would have assumed this code to read the current MTU, deduct the overhead, and then set the MTU to the reduced value.

Comment (jstarks) [inline service/gcsutils/gcstools/netnsConfig.go]: Correct.

Comment (jterry75) [inline service/gcsutils/gcstools/netnsConfig.go]: Good catch. I misread the sample which had a very tricky `-=` not `=` when using `EncapOverhead`

**Comment by jterry75** [comment]
HCS and Docker based tests are working

**Comment by jterry75** [comment]
Any feedback?

**Comment by msabansal** [comment]
I had this in my todo to validate in https://github.com/msabansal/opengcs/tree/encapoverhead Just validated the changes. LGTM

**Comment by msabansal** [comment]
`PS C:\Users\Administrator> docker run -it --rm --net=overlaynet busybox /bin/sh
/ # ifconfig
eth0      Link encap:Ethernet  HWaddr 00:15:5D:1D:77:5D
          inet addr:10.0.0.2  Bcast:0.0.0.0  Mask:255.255.255.0
          inet6 addr: fe80::215:5dff:fe1d:775d/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1450  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:11 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:914 (914.0 B)

eth1      Link encap:Ethernet  HWaddr 00:15:5D:88:8F:87
          inet addr:172.17.26.141  Bcast:0.0.0.0  Mask:255.255.240.0
          inet6 addr: fe80::215:5dff:fe88:8f87/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:696 (696.0 B)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)`

**Comment by jterry75** [comment]
@msabansal - Thanks Sandeep! It took me a very long time to figure out how to test this the first time.

**Pull request merged**